tests/app/main/views/test_index.py

```diff
@@ -64,6 +64,32 @@ def test_landing_page_creates_link_for_document(client, mocker, sample_service):
     )
 
 
+def test_landing_page_creates_link_for_document_with_filename(client, mocker, sample_service):
+    mocker.patch('app.service_api_client.get_service', return_value={'data': sample_service})
+    service_id = uuid4()
+    document_id = uuid4()
+    response = client.get(
+        url_for(
+            'main.landing',
+            service_id=service_id,
+            document_id=document_id,
+            key='1234',
+            filename='custom_file.pdf',
+        )
+    )
+
+    assert response.status_code == 200
+    page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
+
+    assert page.find('a', string="Continue")['href'] == url_for(
+        'main.download_document',
+        service_id=service_id,
+        document_id=document_id,
+        key='1234',
+        filename='custom_file.pdf',
```

Comment:
The important assertion is here, the continue link has got a `filename` query param